src/main/java/core/basesyntax/strategy/impl/PurchaseStrategyImpl.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.strategy.OperationHandler;
+import java.util.function.IntUnaryOperator;
+
+public class PurchaseStrategyImpl implements OperationHandler {
+    private final IntUnaryOperator operation = value -> -value;
+
+    @Override
+    public int calculate(int amount) {
+        return operation.applyAsInt(amount);
+    }
+}
```

Comment:
would also be nice to add a check if we have enough fruit amount in the storage to sell it, and throw ex if we are out of stock on attempt to buy it, and a test for it

Reply:
for this case we have.

![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/73792778/9823f107-838c-4819-9e4e-e8baed0af7cc) and test 
![image](https://github.com/mate-academy/jv-fruit-shop-tests/assets/73792778/29da5d61-3d43-488f-830d-6538a381a25a)
Does the strategy have to know something to the object ?